examples/message/README.md

```diff
@@ -110,10 +107,82 @@ In the future, provider tests will also be included.
 +-------------------+          +-----------+
 ```
 
-## E2E Messaging
+```python
+import pytest
+from pact import MessageProvider
+
+def document_created_handler():
+    return {
+        "event": "ObjectCreated:Put",
+        "documentName": "document.doc",
+        "creator": "TP",
+        "documentType": "microsoft-word"
+    }
+
+
+def test_verify_success():
+    provider = MessageProvider(
+        message_providers={
+            'A document created successfully': document_created_handler
+        },
+        provider='ContentProvider',
+        consumer='DetectContentLambda',
+        pact_dir='pacts'
+
+    )
+    with provider:
+        provider.verify()
+```
+
+
+### Provider with pact broker
+```python
+import pytest
+from pact import MessageProvider
+
+
+PACT_BROKER_URL = "http://localhost"
+PACT_BROKER_USERNAME = "pactbroker"
+PACT_BROKER_PASSWORD = "pactbroker"
+PACT_DIR = "pacts"
+
+
+@pytest.fixture
+def default_opts():
+    return {
+        'broker_username': PACT_BROKER_USERNAME,
+        'broker_password': PACT_BROKER_PASSWORD,
+        'broker_url': PACT_BROKER_URL,
+        'publish_version': '3',
+        'publish_verification_results': False
+    }
+
+def document_created_handler():
+    return {
+        "event": "ObjectCreated:Put",
+        "documentName": "document.doc",
+        "creator": "TP",
+        "documentType": "microsoft-word"
+    }
 
-Note: The current example only tests the consumer side.
-In the future, provider tests will also be included.
+def test_verify_from_broker(default_opts):
+    provider = MessageProvider(
+        message_providers={
+            'A document created successfully': document_created_handler,
+        },
+        provider='ContentProvider',
+        consumer='DetectContentLambda',
```

Comment:
I believe the `consumer` should no longer be mandatory because does not make sense since you fetch the data from the pact broker